infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/LogReplicationEventMetadata.java

```diff
@@ -53,16 +56,31 @@ public LogReplicationEventMetadata(UUID requestId) {
      */
     public LogReplicationEventMetadata(UUID requestId, long syncTimestamp) {
         this.requestId = requestId;
-        this.syncTimestamp = syncTimestamp;
+        this.lastLogEntrySyncedTimestamp = syncTimestamp;
```

Comment:
We better remove this constructor, or add setting the baseSnapshot.

Reply:
For several cases this event only has meaning for log entry and doesn't really handle the baseSnapshot, actually from where we call it, it has no notion of it.